src/sage/groups/libgap_mixin.py

```diff
@@ -987,113 +987,213 @@ def minimum_generating_set(G) -> list:
+        `N_{i_1},N_{i_2}\dots N_{i_t} N_{i_{t+1}} \in \bold{N}`,
+        Compute `g^* =`
+        `\{g_1N_{i_1},g_{i_2}N_{i_3}\dots`
+        `g_{i_t}N_t,g_{t+1}\dots g_s\}`.
+        If `\braket{g^*G_i}\; = G/G_i`, return `{g^*}`
+
+        By now, we must have exhausted our search.
+
+        """
+        def gen_combinations(g: list, N: list, t: int):
+            r"""
+            Generates all modifications to `g` of form
+            `g^* = \{g_1 N_1,g_2 N_2 \dots g_t N_t , g_{t+1}, g_{t+2} \dots\}`
+
+            INPUT:
+
+                - ``g`` -- a list of elements from some group `G`
+
+                - ``N`` -- a list of elements of
+                  normal subgroup `N` of `G`
+
+                - ``t`` -- the index till which to modify ``g``
+            """
+            if t == 0:
+                yield g
+                return
+            for gm in gen_combinations(g, N, t-1):
+                for n in N:
+                    old = gm[t-1]
+                    gm[t-1] = old * n
+                    yield gm
+                    gm[t-1] = old
+        s = len(G_by_Gim1_mingen_reps)
+        Gim1_by_Gi_L = list(Gim1_by_Gi.AsList())
+        Gim1_by_Gi_elem_reps = [
+            phi_Gim1_by_Gi.PreImagesRepresentative(x) for x in Gim1_by_Gi_L]
+        Gim1_by_Gi_gen = list(libgap.SmallGeneratingSet(Gim1_by_Gi))
+        Gim1_by_Gi_gen_reps = [
+            phi_Gim1_by_Gi.PreImagesRepresentative(x) for x in Gim1_by_Gi_gen]
+        if Gim1_by_Gi.IsAbelian().sage():
+            if (G_by_Gi == libgap.GroupByGenerators(
+                [phi_G_by_Gi.ImagesRepresentative(x)
+                 for x in G_by_Gim1_mingen_reps])):
+                return G_by_Gim1_mingen_reps
+            for i in range(s):
+                for j in range(len(Gim1_by_Gi_gen_reps)):
+                    temp = G_by_Gim1_mingen_reps[i]
+                    G_by_Gim1_mingen_reps[i] = G_by_Gim1_mingen_reps[i] * \
+                        Gim1_by_Gi_gen_reps[j]
+                    if (G_by_Gi == libgap.GroupByGenerators(
+                        [phi_G_by_Gi.ImagesRepresentative(x)
+                         for x in G_by_Gim1_mingen_reps])):
+                        return G_by_Gim1_mingen_reps
+                    G_by_Gim1_mingen_reps[i] = temp
+            return G_by_Gim1_mingen_reps + [Gim1_by_Gi_gen_reps[0]]
+        for raw_gens in gen_combinations(
+                G_by_Gim1_mingen_reps,
+                Gim1_by_Gi_elem_reps, s):
+            if (G_by_Gi == libgap.GroupByGenerators(
+                [phi_G_by_Gi.ImagesRepresentative(x)
+                 for x in raw_gens])):
+                return raw_gens
+        for raw_gens in gen_combinations(
+                G_by_Gim1_mingen_reps+[Gim1_by_Gi_elem_reps[0]],
+                Gim1_by_Gi_elem_reps, s+1):
+            if (G_by_Gi == libgap.GroupByGenerators(
+                [phi_G_by_Gi.ImagesRepresentative(x)
+                 for x in raw_gens])):
+                return raw_gens
```

Comment:
this is not according to the pep8 rules.

Reply:
What should I change ?